firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -407,8 +412,16 @@ private static void logBadConfigError() {
             + " initializing Firebase.");
   }
 
-  private HttpURLConnection openHttpURLConnection(URL url, String apiKey) throws IOException {
-    HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection();
+  private HttpURLConnection openHttpURLConnection(URL url, String apiKey)
+      throws FirebaseInstallationsException {
+    HttpURLConnection httpURLConnection;
+    try {
+      httpURLConnection = (HttpURLConnection) url.openConnection();
```

Comment:
Why are you casting this URLConnection?
Are you using methods that are specific to HttpURLConnection?
If not, I strongly recommend programming against the interface than the implementation.

Reply:
Yes, I am using methods specific to HttpURLConnection